lib/backend.js

```diff
@@ -789,7 +791,7 @@ Backend.prototype._fetchSnapshot = function(collection, id, version, callback) {
     // - we want to avoid the 'op' middleware, because we later use the 'readSnapshots' middleware in _sanitizeSnapshots
     // - we handle the projection in _sanitizeSnapshots
     var from = milestoneSnapshot ? milestoneSnapshot.v : 0;
-    db.getOps(collection, id, from, version, null, function(error, ops) {
+    db.getOps(collection, id, from, version, options, function(error, ops) {
```

Comment:
I'm not sure we've put this in the right place? Shouldn't we be fetching op metadata in [`_getSanitizedOps()`](https://github.com/share/sharedb/blob/cc0e3382bf3df4c38c7ccc6bc52da71aff8c9f74/lib/backend.js#L327) (and maybe its [bulk counterpart](https://github.com/share/sharedb/blob/cc0e3382bf3df4c38c7ccc6bc52da71aff8c9f74/lib/backend.js#L336))?

Reply:
It seems that if a snapshot has some saved metadata and we do not pass `fields` and `options` to  [getSnapshot](https://github.com/share/sharedb-mongo/blob/e72c6bb1a3e9896b29a3f0cbb8f120fe5d53bd63/index.js#L335) , the metadata will not be retrieved at all. Same with `_getSanitizedOps`, if we do not pass the `options` to [getOps](https://github.com/share/sharedb/blob/cc0e3382bf3df4c38c7ccc6bc52da71aff8c9f74/lib/backend.js#L369) , the database layer will not retrieve the metadata and there will be nothing to sanitize. I have a very shallow understanding of the intent here, and maybe I am missing something fundamental. I could definitely use your help to understand this better. Thank you!

Reply:
I think we should basically not touch snapshots at all in this PR. As I mentioned in my [other comment](https://github.com/share/sharedb/pull/586#discussion_r1091765989), snapshot metadata is a very different beast, and I think we should consider it separately to ops.

I *think* this belongs in `_getSanitizedOps()` with the call to [`backend.db.getOps()`](https://github.com/share/sharedb/blob/cc0e3382bf3df4c38c7ccc6bc52da71aff8c9f74/lib/backend.js#L327).